crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -436,51 +444,88 @@ impl<'a> Submitter<'a> {
                 Err(err) => match err {
                     SubmitApiError::InvalidNonce => {
-                        tracing::warn!("submission failed: invalid nonce")
+                        tracing::warn!("{} submission failed: invalid nonce", submitter_name)
                     }
                     SubmitApiError::ReplacementTransactionUnderpriced => {
-                        tracing::warn!("submission failed: replacement transaction underpriced")
+                        tracing::warn!(
+                            "{} submission failed: replacement transaction underpriced",
+                            submitter_name
+                        )
                     }
                     SubmitApiError::OpenEthereumTooCheapToReplace => {
-                        tracing::debug!("submission failed: OE has different replacement rules than our algorithm")
+                        tracing::debug!("{} submission failed: OE has different replacement rules than our algorithm", submitter_name)
                     }
                     SubmitApiError::EdenTransactionTooExpensive => {
-                        tracing::warn!("submission failed: eden transaction too expensive")
+                        tracing::warn!(
+                            "{} submission failed: eden transaction too expensive",
+                            submitter_name
+                        )
+                    }
+                    SubmitApiError::Other(err) => {
+                        tracing::error!("{} submission failed: {:?}", submitter_name, err)
                     }
-                    SubmitApiError::Other(err) => tracing::error!("submission failed: {:?}", err),
                 },
             }
             tokio::time::sleep(params.retry_interval).await;
         }
     }
 
     /// Prepare transaction for simulation
-    fn build_method(
+    async fn build_method(
         &self,
         settlement: Settlement,
         gas_price: &EstimatedGasPrice,
         nonce: U256,
         gas_limit: f64,
     ) -> MethodBuilder<DynTransport, ()> {
-        settle_method_builder(self.contract, settlement.into(), self.account.clone())
+        let method = settle_method_builder(self.contract, settlement.into(), self.account.clone())
             .nonce(nonce)
             .gas(U256::from_f64_lossy(gas_limit))
-            .gas_price(crate::into_gas_price(gas_price))
+            .gas_price(crate::into_gas_price(gas_price));
+        match self.estimate_access_list(&method.tx).await {
+            Ok(access_list) => method.access_list(access_list),
+            Err(err) => {
+                tracing::warn!("access list failure: {:?}", err);
+                method
+            }
+        }
+    }
+
+    /// Estimate access list and validate
+    async fn estimate_access_list(
+        &self,
+        tx: &TransactionBuilder<DynTransport>,
+    ) -> Result<AccessList> {
+        let access_list = self.access_list_estimator.estimate_access_list(tx).await?;
```

Comment:
Maybe the sanitization of the access list should also move here? This way each implementation doesn't need to call it themselves.

Reply:
Hm, then I would have to add the sanitization for settlement simulation also, so it's not that beneficial, What we can do is to add the sanitization to `PriorityAccessListEstimating` but also make `NodeApi` and `TenderlyApi` private so we are forced to use `PriorityAccessListEstimating`

Reply:
Fixed as explained.
Also feels like filtering is the response of the estimator, not the client.